CLN: Deprecate dayofweek/hello day_of_week (#9606)

abhishekmangla opened the pull request:
Rename dayofweek properties in Period, Datetime, and PeriodArray classes.
Add day_of_week property to respective tests for each class.

- [x] closes #9606
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

abhishekmangla committed:
CLN: Deprecate dayofweek/hello day_of_week (#9606)

Rename dayofweek properties in Period, Datetime, and PeriodArray classes.
Add day_of_week property to respective tests for each class.

abhishekmangla committed:
Test precommit hook

abhishekmangla committed:
Retest precommit

abhishekmangla committed:
Fix whitespace

arw2019 commented on pandas/_libs/tslibs/timestamps.pyx:
```diff
@@ -538,12 +538,19 @@ cdef class _Timestamp(ABCTimestamp):
         """
         return self.weekday()
 
+    @property
```
You don't need to define a second property, set `dayofweek` to be an alias of `day_of_week`

arw2019 commented on pandas/_libs/tslibs/period.pyx:
```diff
@@ -1900,38 +1900,43 @@ cdef class _Period(PeriodMixin):
         base = self._dtype._dtype_code
         return pweekday(self.ordinal, base)
 
+    @property
```
I don't think you need this? Just set dayofweek` to be an alias of `day_of_week`

arw2019 commented on pandas/tests/generic/test_finalize.py:
```diff
@@ -678,6 +678,8 @@ def test_datetime_method(method):
         "microsecond",
         "nanosecond",
         "dayofweek",
+        # Add better named field for GH-9606
```
nit-pick but we usually don't leave comments like this: git/blame keeps track of when changes occurred

arw2019 commented on pandas/tests/indexes/datetimes/test_scalar_compat.py:
```diff
@@ -38,6 +38,8 @@ def test_dti_date_out_of_range(self, data):
         "field",
         [
             "dayofweek",
+            # Add better named field for GH-9606
```
same here

arw2019 commented on pandas/tests/indexes/period/test_period.py:
```diff
@@ -249,6 +249,8 @@ def _check_all_fields(self, periodindex):
             "weekofyear",
             "week",
             "dayofweek",
+            # Add better named field for GH-9606
```
same here

arw2019 reviewed (changes requested):
Thanks @abhishekmangla for the PR!

Looks good, some comments.

You'll also need to add a whatsnew (v1.2) under enhancements

Let us know if you have any questions!

abhishekmangla committed:
CLN: Resolve comments from @arw2019

Alias dep. function instead of extra property
Remove comments from test files

abhishekmangla committed:
Add proposed change to whatsnew

abhishekmangla commented:
@arw2019 Thank you for the review. I addressed your comments and added an entry to whatsnew file.

arw2019 commented on doc/source/whatsnew/v1.2.0.rst:
```diff
@@ -13,6 +13,21 @@ including other versions of pandas.
 Enhancements
 ~~~~~~~~~~~~
 
+.. _whatsnew_120.improve_timeseries_accessor_naming:
```
sorry to be a pain but could you do something short here like the whatsnew in this PR:
 https://github.com/pandas-dev/pandas/pull/9605/files

arw2019 commented on pandas/_libs/tslibs/period.pyx:
```diff
@@ -1475,6 +1475,9 @@ cdef class _Period(PeriodMixin):
         PeriodDtypeBase _dtype
         BaseOffset freq
 
+    # Alias dayofweek function to day_of_week (GH-9606)
```
let's drop this comment

arw2019 commented on pandas/_libs/tslibs/timestamps.pyx:
```diff
@@ -230,6 +230,8 @@ cdef class _Timestamp(ABCTimestamp):
 
     # higher than np.ndarray and np.matrix
     __array_priority__ = 100
+    # Alias dayofweek function to day_of_week (GH-9606)
```
Same here, let's drop this comment

arw2019 reviewed (changes requested):
Thanks for following up on this!

Some more (minor) comments, otherwise good to go IMO

abhishekmangla committed:
ENH: Support day_of_year over dayofyear (#9606)

Similar to changes for day_of_week, create aliases for day_of_year and add to respective test files to ensure back-compat.
Also fix whatsnew rst file

abhishekmangla committed:
Add autogenerated doc references

abhishekmangla commented:
@arw2019 Resolved minor issues. Do you know how to locally test .rst files, in the CI stage of the checks, I am failing things like
`/home/runner/work/pandas/pandas/pandas/core/indexes/period.py:docstring of pandas.PeriodIndex.rst:77: WARNING: autosummary: stub file not found 'pandas.PeriodIndex.day_of_year'. Check your autosummary_generate setting.
2391`
AND
`/home/runner/work/pandas/pandas/doc/source/whatsnew/v1.2.0.rst:31: WARNING: Explicit markup ends without a blank line; unexpected unindent.`

`./ci/code_checks.sh` shows no errors locally

arw2019 commented:
> @arw2019 Resolved minor issues. Do you know how to locally test .rst files? In the CI stage of the checks, I am failing things like
> `/home/runner/work/pandas/pandas/pandas/core/indexes/period.py:docstring of pandas.PeriodIndex.rst:77: WARNING: autosummary: stub file not found 'pandas.PeriodIndex.day_of_year'. Check your autosummary_generate setting. 2391`

I think you (maybe) need to add the new methods to the rst files manually - for example in doc/source/reference/series.rst both    daysinmonth and days_in_month are listed

> `/home/runner/work/pandas/pandas/doc/source/whatsnew/v1.2.0.rst:31: WARNING: Explicit markup ends without a blank line; unexpected unindent.`

This one is a linting issue

abhishekmangla committed:
Fix newline error Sphinx in whatsnew doc

jreback commented on doc/source/whatsnew/v1.2.0.rst:
```diff
@@ -13,14 +13,23 @@ including other versions of pandas.
 
+
+
+.. _whatsnew_120.improve_timeseries_accessor_naming:
```
you can just put these notes in Other Enhancements section

jreback commented on doc/source/whatsnew/v1.2.0.rst:
```diff
@@ -13,14 +13,23 @@ including other versions of pandas.
-:class:`Series` and :class:`DataFrame` can now be created with ``allows_duplicate_labels=False`` flag to
-control whether the index or columns can contain duplicate labels (:issue:`28394`). This can be used to
-prevent accidental introduction of duplicate labels, which can affect downstream operations.
+:class:`Series` and :class:`DataFrame` can now be created with ``allows_duplicate_labels=False`` flag to control whether the index or columns can contain duplicate labels (:issue:`28394`). This can be used to prevent accidental introduction of duplicate labels, which can affect downstream operations.
```
pls don't adjust other notes

abhishekmangla commented:
@arw2019 The only unit test that is failing on Azure pipeline is `test_binary_arith_ops` which seems to be a known unreliable test (#37328). (The test suite passes on my macOS machine)

Is this good to merge now?

jreback reviewed (changes requested):
this is fine. ping on green.

jreback commented:
@pandas-dev/pandas-core should we do a DeprecationWarning / FutureWarning on the original attributes? (would be a followon)

abhishekmangla committed:
Move whatsnew info to Other Enhancements

abhishekmangla commented:
@jreback All checks green except aforementioned `test_binary_arith_ops`

jreback commented:
thanks @abhishekmangla very nice
